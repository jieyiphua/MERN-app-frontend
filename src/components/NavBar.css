.navBarContainer {
  height: 100px;
  width: 100vw;
  background-color: red;
  display: flex;
  justify-content: flex-start;
  align-content: center;
}

.logo {
  height: 90%;
}
