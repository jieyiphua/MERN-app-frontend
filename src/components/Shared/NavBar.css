.nav-bar {
  height: 130px;
  width: 100vw;
  background-color: white;
  display: flex;
  justify-content: space-between;
  align-content: center;
  position: fixed;
  z-index: 3;
  /* box-shadow: 0 2px 2px -2px rgba(0, 0, 0, 0.2); */
}

.nav-bar-right-box {
  display: flex;
  flex-direction: row;
}

.company-logo {
  width: 170px;
  height: 100%;
  background-image: url("./CompanyLogo.jpg");
  background-size: contain;
  background-repeat: no-repeat;
  margin-left: 20px;
}
/* 
.nav-bar-links {
  text-decoration: none;
  color: black;
  font-size: 1.5vw;
} */

/* .nav-bar-link-box {
  width: 12vw;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
} */
/* 
.log-out-button {
  width: 50px;
  height: 50px;
  background: rgb(248, 121, 121);
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  color: white;
  padding: 5px 5px;
  margin: 10px 10px;
  font-size: 70%;
  font-weight: 800;
  cursor: pointer;
} */
