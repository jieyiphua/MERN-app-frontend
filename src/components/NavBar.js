import "./NavBar.css";

import React, { Component } from "react";

class NavBar extends Component {
  render() {
    return (
      <div className="navBarContainer">
        <svg
          className="logo"
          width="603"
          height="278"
          viewBox="0 0 603 278"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <ellipse cx="181.5" cy="145" rx="17.5" ry="18" fill="white" />
          <path
            d="M193.964 121.48C199.964 121.48 205.404 122.88 210.284 125.68C215.244 128.4 219.124 132.28 221.924 137.32C224.724 142.28 226.124 148.08 226.124 154.72C226.124 161.36 224.724 167.2 221.924 172.24C219.124 177.2 215.244 181.08 210.284 183.88C205.404 186.6 199.964 187.96 193.964 187.96C185.084 187.96 178.324 185.16 173.684 179.56V187H155.804V97.96H174.524V129.28C179.244 124.08 185.724 121.48 193.964 121.48ZM190.724 172.6C195.524 172.6 199.444 171 202.484 167.8C205.604 164.52 207.164 160.16 207.164 154.72C207.164 149.28 205.604 144.96 202.484 141.76C199.444 138.48 195.524 136.84 190.724 136.84C185.924 136.84 181.964 138.48 178.844 141.76C175.804 144.96 174.284 149.28 174.284 154.72C174.284 160.16 175.804 164.52 178.844 167.8C181.964 171 185.924 172.6 190.724 172.6ZM257.856 187.96C250.976 187.96 244.776 186.56 239.256 183.76C233.816 180.88 229.536 176.92 226.416 171.88C223.376 166.84 221.856 161.12 221.856 154.72C221.856 148.32 223.376 142.6 226.416 137.56C229.536 132.52 233.816 128.6 239.256 125.8C244.776 122.92 250.976 121.48 257.856 121.48C264.656 121.48 270.576 122.92 275.616 125.8C280.736 128.6 284.456 132.64 286.776 137.92L272.256 145.72C268.896 139.8 264.056 136.84 257.736 136.84C252.856 136.84 248.816 138.44 245.616 141.64C242.416 144.84 240.816 149.2 240.816 154.72C240.816 160.24 242.416 164.6 245.616 167.8C248.816 171 252.856 172.6 257.736 172.6C264.136 172.6 268.976 169.64 272.256 163.72L286.776 171.64C284.456 176.76 280.736 180.76 275.616 183.64C270.576 186.52 264.656 187.96 257.856 187.96ZM368.234 121.48C376.314 121.48 382.714 123.88 387.434 128.68C392.234 133.4 394.634 140.52 394.634 150.04V187H375.914V152.92C375.914 147.8 374.834 144 372.674 141.52C370.594 138.96 367.594 137.68 363.674 137.68C359.274 137.68 355.794 139.12 353.234 142C350.674 144.8 349.394 149 349.394 154.6V187H330.674V152.92C330.674 142.76 326.594 137.68 318.434 137.68C314.114 137.68 310.674 139.12 308.114 142C305.554 144.8 304.274 149 304.274 154.6V187H285.554V122.44H303.434V129.88C305.834 127.16 308.754 125.08 312.194 123.64C315.714 122.2 319.554 121.48 323.714 121.48C328.274 121.48 332.394 122.4 336.074 124.24C339.754 126 342.714 128.6 344.954 132.04C347.594 128.68 350.914 126.08 354.914 124.24C358.994 122.4 363.434 121.48 368.234 121.48ZM425.453 121.48C435.453 121.48 443.133 123.88 448.493 128.68C453.853 133.4 456.533 140.56 456.533 150.16V187H439.013V178.96C435.493 184.96 428.933 187.96 419.333 187.96C414.373 187.96 410.053 187.12 406.373 185.44C402.773 183.76 400.013 181.44 398.093 178.48C396.173 175.52 395.213 172.16 395.213 168.4C395.213 162.4 397.453 157.68 401.933 154.24C406.493 150.8 413.493 149.08 422.933 149.08H437.813C437.813 145 436.573 141.88 434.093 139.72C431.613 137.48 427.893 136.36 422.933 136.36C419.493 136.36 416.093 136.92 412.733 138.04C409.453 139.08 406.653 140.52 404.333 142.36L397.613 129.28C401.133 126.8 405.333 124.88 410.213 123.52C415.173 122.16 420.253 121.48 425.453 121.48ZM424.013 175.36C427.213 175.36 430.053 174.64 432.533 173.2C435.013 171.68 436.773 169.48 437.813 166.6V160H424.973C417.293 160 413.453 162.52 413.453 167.56C413.453 169.96 414.373 171.88 416.213 173.32C418.133 174.68 420.733 175.36 424.013 175.36Z"
            fill="#DDDDDD"
          />
          <path
            d="M184.964 111.48C190.964 111.48 196.404 112.88 201.284 115.68C206.244 118.4 210.124 122.28 212.924 127.32C215.724 132.28 217.124 138.08 217.124 144.72C217.124 151.36 215.724 157.2 212.924 162.24C210.124 167.2 206.244 171.08 201.284 173.88C196.404 176.6 190.964 177.96 184.964 177.96C176.084 177.96 169.324 175.16 164.684 169.56V177H146.804V87.96H165.524V119.28C170.244 114.08 176.724 111.48 184.964 111.48ZM181.724 162.6C186.524 162.6 190.444 161 193.484 157.8C196.604 154.52 198.164 150.16 198.164 144.72C198.164 139.28 196.604 134.96 193.484 131.76C190.444 128.48 186.524 126.84 181.724 126.84C176.924 126.84 172.964 128.48 169.844 131.76C166.804 134.96 165.284 139.28 165.284 144.72C165.284 150.16 166.804 154.52 169.844 157.8C172.964 161 176.924 162.6 181.724 162.6ZM248.856 177.96C241.976 177.96 235.776 176.56 230.256 173.76C224.816 170.88 220.536 166.92 217.416 161.88C214.376 156.84 212.856 151.12 212.856 144.72C212.856 138.32 214.376 132.6 217.416 127.56C220.536 122.52 224.816 118.6 230.256 115.8C235.776 112.92 241.976 111.48 248.856 111.48C255.656 111.48 261.576 112.92 266.616 115.8C271.736 118.6 275.456 122.64 277.776 127.92L263.256 135.72C259.896 129.8 255.056 126.84 248.736 126.84C243.856 126.84 239.816 128.44 236.616 131.64C233.416 134.84 231.816 139.2 231.816 144.72C231.816 150.24 233.416 154.6 236.616 157.8C239.816 161 243.856 162.6 248.736 162.6C255.136 162.6 259.976 159.64 263.256 153.72L277.776 161.64C275.456 166.76 271.736 170.76 266.616 173.64C261.576 176.52 255.656 177.96 248.856 177.96ZM359.234 111.48C367.314 111.48 373.714 113.88 378.434 118.68C383.234 123.4 385.634 130.52 385.634 140.04V177H366.914V142.92C366.914 137.8 365.834 134 363.674 131.52C361.594 128.96 358.594 127.68 354.674 127.68C350.274 127.68 346.794 129.12 344.234 132C341.674 134.8 340.394 139 340.394 144.6V177H321.674V142.92C321.674 132.76 317.594 127.68 309.434 127.68C305.114 127.68 301.674 129.12 299.114 132C296.554 134.8 295.274 139 295.274 144.6V177H276.554V112.44H294.434V119.88C296.834 117.16 299.754 115.08 303.194 113.64C306.714 112.2 310.554 111.48 314.714 111.48C319.274 111.48 323.394 112.4 327.074 114.24C330.754 116 333.714 118.6 335.954 122.04C338.594 118.68 341.914 116.08 345.914 114.24C349.994 112.4 354.434 111.48 359.234 111.48ZM416.453 111.48C426.453 111.48 434.133 113.88 439.493 118.68C444.853 123.4 447.533 130.56 447.533 140.16V177H430.013V168.96C426.493 174.96 419.933 177.96 410.333 177.96C405.373 177.96 401.053 177.12 397.373 175.44C393.773 173.76 391.013 171.44 389.093 168.48C387.173 165.52 386.213 162.16 386.213 158.4C386.213 152.4 388.453 147.68 392.933 144.24C397.493 140.8 404.493 139.08 413.933 139.08H428.813C428.813 135 427.573 131.88 425.093 129.72C422.613 127.48 418.893 126.36 413.933 126.36C410.493 126.36 407.093 126.92 403.733 128.04C400.453 129.08 397.653 130.52 395.333 132.36L388.613 119.28C392.133 116.8 396.333 114.88 401.213 113.52C406.173 112.16 411.253 111.48 416.453 111.48ZM415.013 165.36C418.213 165.36 421.053 164.64 423.533 163.2C426.013 161.68 427.773 159.48 428.813 156.6V150H415.973C408.293 150 404.453 152.52 404.453 157.56C404.453 159.96 405.373 161.88 407.213 163.32C409.133 164.68 411.733 165.36 415.013 165.36Z"
            fill="black"
          />
          <rect x="146" y="26" width="20" height="151" fill="black" />
          <path
            d="M166 139L168.5 137.5L169.5 151.5C168.833 153.167 167.2 155.3 166 150.5C164.8 145.7 165.5 140.833 166 139Z"
            fill="#DDDDDD"
          />
        </svg>
      </div>
    );
  }
}

export default NavBar;
