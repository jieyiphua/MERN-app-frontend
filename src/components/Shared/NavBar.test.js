import NavBar from "./NavBar";
import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import sinon from "sinon";

describe("<NavBar />", () => {
  it("renders component", () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
