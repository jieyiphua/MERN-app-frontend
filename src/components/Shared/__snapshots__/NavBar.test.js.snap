// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders component 1`] = `
<div
  className="nav-bar"
>
  <Link
    className="company-logo"
    to="/"
  />
  <div
    className="nav-bar-right-box"
  >
    <Bootstrap(undefined)
      bg="light"
      expand="lg"
      id="nav-background"
    >
      <NavbarToggle
        aria-controls="basic-navbar-nav"
        label="Toggle navigation"
      />
      <NavbarCollapse
        id="basic-navbar-nav"
      >
        <Nav
          className="mr-auto"
          fill={false}
          justify={false}
        >
          <NavLink
            as={[Function]}
            disabled={false}
            href="/"
          >
            Home
          </NavLink>
          <NavDropdown
            id="basic-nav-dropdown"
            title="About"
          >
            <DropdownItem
              as={[Function]}
              disabled={false}
              href="/about/our-story"
            >
              About us
            </DropdownItem>
            <DropdownItem
              as={[Function]}
              disabled={false}
              href="/about/instructors"
            >
              Faculty
            </DropdownItem>
            <DropdownItem
              as={[Function]}
              disabled={false}
              href="/faqs"
            >
              FAQs
            </DropdownItem>
          </NavDropdown>
          <NavLink
            as={[Function]}
            disabled={false}
            href="/classes"
          >
            Classes
          </NavLink>
          <NavLink
            as={[Function]}
            disabled={false}
            href="/about/gallery"
          >
            Gallery
          </NavLink>
          <NavLink
            as={[Function]}
            disabled={false}
            href="/whats-on/events"
          >
            What's on
          </NavLink>
          <NavLink
            as={[Function]}
            disabled={false}
            href="/visit/contact"
          >
            Contact us
          </NavLink>
          <NavLink
            as={[Function]}
            disabled={false}
            href="https://www.facebook.com/ballaratcentreofmusicandthearts/"
            target="none"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f09a",
                    "M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z",
                  ],
                  "iconName": "facebook",
                  "prefix": "fab",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </NavLink>
        </Nav>
      </NavbarCollapse>
    </Bootstrap(undefined)>
  </div>
</div>
`;
